File: packages/php-sdk/src/Types/Requests/Quote/UpdateContactRequest.php
<?php

declare(strict_types=1);

namespace TurboDocx\Types\Requests\Quote;

/**
 * Request for updating a contact
 */
final class UpdateContactRequest
{
    public function __construct(
        public readonly ?string $name = null,
        public readonly ?string $email = null,
        public readonly ?string $phone = null,
        public readonly ?string $title = null,
    ) {}

    /**
     * @return array<string, mixed>
     */
    public function toArray(): array
    {
        $data = [];

        if ($this->name !== null) {
            $data['name'] = $this->name;
        }
        if ($this->email !== null) {
            $data['email'] = $this->email;
        }
        if ($this->phone !== null) {
            $data['phone'] = $this->phone;
        }
        if ($this->title !== null) {
            $data['title'] = $this->title;
        }

        return $data;
    }
}

File: packages/java-sdk/src/main/java/com/turbodocx/models/quote/UpdateContactRequest.java
package com.turbodocx.models.quote;

/**
 * Request to update a contact.
 */
public class UpdateContactRequest extends TrackableRequest {
    private String name;
    private String email;
    private String phone;
    private String title;

    public String getName() { return name; }
    public void setName(String name) { this.name = name; markFieldSet("name"); }
    public String getEmail() { return email; }
    public void setEmail(String email) { this.email = email; markFieldSet("email"); }
    public String getPhone() { return phone; }
    public void setPhone(String phone) { this.phone = phone; markFieldSet("phone"); }
    public String getTitle() { return title; }
    public void setTitle(String title) { this.title = title; markFieldSet("title"); }
}
